Support databricks_tags for MV/STs

ericj-db opened the pull request:
<!-- Please review our pull request review process in CONTRIBUTING.md before your proceed. -->

Resolves #

<!---
  Include the number of the issue addressed by this PR above if applicable.

  Example:
    resolves #1234

  Please review our pull request review process in CONTRIBUTING.md before your proceed.
-->

### Description

- Reuse the existing `apply_tags` macro for MV/STs and add functional tests for validation
- Add a new `DatabricksRelationType.render()` function. We are using `ALTER {{ relation.type }}` in many existing macros. This does not work for types that have underscores like materialized_view and streaming_table. Updated the macros to use `ALTER {{ relation.type.render() }}` instead

### Checklist

- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-databricks next" section.

ericj-db commented on dbt/include/databricks/macros/adapters/persist_docs.sql:
```diff
@@ -28,7 +28,7 @@ COMMENT ON COLUMN {{ column_path }} IS '{{ escaped_comment }}'
 
 {% macro alter_relation_comment_sql(relation, description) %}
-COMMENT ON {{ relation.type.upper() }} {{ relation.render() }} IS '{{ description | replace("'", "\\'") }}'
+COMMENT ON {{ relation.type.render().upper() }} {{ relation.render() }} IS '{{ description | replace("'", "\\'") }}'
```
I don't think that's safe because of other usages like https://github.com/databricks/dbt-databricks/blob/92f1442faabe0fce6f0375b95e46ebcbfcea4c67/dbt/include/databricks/macros/relations/components/column_tags.sql#L38-L40

benc-db replied:
I meant specifically for relation.type.render, which we have control over.

ericj-db replied:
Oh yeah I mean line 38, we're doing `relation.type == 'view'`. We would need to update it to `relation.type == 'VIEW'`, right?

benc-db replied:
I mean line 31.  You have relation.type.render().upper().  You introduced relation type render in this PR, so I'm wondering why not include upper inside its definition.

ericj-db replied:
Sorry, I'm kind of combining your 2 suggestions in this thread and maybe that's causing some confusion. If we apply both suggestions, we get this
```
DatabricksRelationType(StrEnum):
  def __str__(self):
    return self.value.replace("_", " ").upper()
```

My point is that we are not only using `relation.type` as a drop-in for `ALTER {{ relation.type }} ...` statements. We are also using it in places like https://github.com/databricks/dbt-databricks/blob/92f1442faabe0fce6f0375b95e46ebcbfcea4c67/dbt/include/databricks/macros/relations/components/column_tags.sql#L38-L40 

(I'm referring to line 38 in this snippet). We would need to update that to `{%- if relation.type == 'VIEW' -%} `. Is that what we want? I wasn't sure if this is a risky change if macros outside of this package rely on `relation.type`

benc-db replied:
ah, yeah, this is a reason to not override the to string, but we can move the upper into render.

ericj-db replied:
Addressed in a9bfe9fc4438ed14ee25c8be47be164df54704b5

ericj-db commented on dbt/adapters/databricks/relation.py:
```diff
@@ -43,6 +43,10 @@ class DatabricksRelationType(StrEnum):
     MetricView = "metric_view"
     Unknown = "unknown"
 
+    def render(self) -> str:
```
Same as https://github.com/databricks/dbt-databricks/pull/1142#discussion_r2271159638, it will affect other macro usages

ericj-db commented:
> Don't we want to use the relation config machinery to ensure we detect changes?

Yeah my bad, totally missed this. Addressed this and added functional tests for applying changes in 3e7ca842faa87a86e7e7285bf5d9a33ace8e17cf
